File: rerun_cpp/src/rerun/components/video_timestamp_ext.cpp
#if 0

// <CODEGEN_COPY_TO_HEADER>
#include <chrono>
// </CODEGEN_COPY_TO_HEADER>

#include "../datatypes/video_time_mode.hpp"
#include "video_timestamp.hpp"

namespace rerun {
    namespace components {

        // <CODEGEN_COPY_TO_HEADER>

        /// Creates a new `VideoTimestamp` from a presentation timestamp as a chrono duration.
        template <typename TRep, typename TPeriod>
        VideoTimestamp(std::chrono::duration<TRep, TPeriod> time) {
            timestamp.timestamp_ns =
                std::chrono::duration_cast<std::chrono::nanoseconds>(time).count();
        }

        /// Creates a new `VideoTimestamp` from a presentation timestamp in seconds.
        static VideoTimestamp from_secs(double seconds) {
            return VideoTimestamp(std::chrono::duration<double>(seconds));
        }

        /// Creates a new `VideoTimestamp` from a presentation timestamp in seconds.
        [[deprecated("Renamed to `from_secs`")]]
        static VideoTimestamp from_seconds(double seconds) {
            return VideoTimestamp(std::chrono::duration<double>(seconds));
        }

        /// Creates a new `VideoTimestamp` from a presentation timestamp in milliseconds.
        static VideoTimestamp from_millis(double milliseconds) {
            return VideoTimestamp(std::chrono::duration<double, std::milli>(milliseconds));
        }

        /// Creates a new `VideoTimestamp` from a presentation timestamp in milliseconds.
        [[deprecated("Renamed to `from_millis`")]]
        static VideoTimestamp from_milliseconds(double milliseconds) {
            return VideoTimestamp(std::chrono::duration<double, std::milli>(milliseconds));
        }

        /// Creates a new `VideoTimestamp` from a presentation timestamp in nanoseconds.
        static VideoTimestamp from_nanos(int64_t nanoseconds) {
            return VideoTimestamp(std::chrono::nanoseconds(nanoseconds));
        }

        /// Creates a new `VideoTimestamp` from a presentation timestamp in nanoseconds.
        [[deprecated("Renamed to `from_nanos`")]]
        static VideoTimestamp from_nanoseconds(int64_t nanoseconds) {
            return VideoTimestamp(std::chrono::nanoseconds(nanoseconds));
        }

        // </CODEGEN_COPY_TO_HEADER>

    } // namespace components
} // namespace rerun

#endif

File: crates/store/re_types/src/components/video_timestamp_ext.rs
use super::VideoTimestamp;

impl VideoTimestamp {
    /// Create new timestamp from seconds since video start.
    #[inline]
    pub fn from_secs(seconds: f64) -> Self {
        crate::datatypes::VideoTimestamp::from_nanos((seconds * 1e9).round() as i64).into()
    }

    /// Create new timestamp from seconds since video start.
    #[deprecated(since = "0.23.0", note = "Use `from_secs` instead")]
    #[inline]
    pub fn from_seconds(seconds: f64) -> Self {
        crate::datatypes::VideoTimestamp::from_nanos((seconds * 1e9).round() as i64).into()
    }

    /// Create new timestamp from milliseconds since video start.
    #[inline]
    pub fn from_millis(milliseconds: f64) -> Self {
        crate::datatypes::VideoTimestamp::from_nanos((milliseconds * 1e6).round() as i64).into()
    }

    /// Create new timestamp from milliseconds since video start.
    #[deprecated(since = "0.23.0", note = "Use `from_millis` instead")]
    #[inline]
    pub fn from_milliseconds(milliseconds: f64) -> Self {
        crate::datatypes::VideoTimestamp::from_nanos((milliseconds * 1e6).round() as i64).into()
    }

    /// Create new timestamp from nanoseconds since video start.
    #[inline]
    pub fn from_nanos(nanos: i64) -> Self {
        crate::datatypes::VideoTimestamp::from_nanos(nanos).into()
    }

    /// Create new timestamp from nanoseconds since video start.
    #[deprecated(since = "0.23.0", note = "Use `from_nanos` instead")]
    #[inline]
    pub fn from_nanoseconds(nanos: i64) -> Self {
        crate::datatypes::VideoTimestamp::from_nanos(nanos).into()
    }
}
